opencog/nlp/learn/link-pipeline.scm

```diff
@@ -128,12 +128,7 @@
 ; it parsed, and then updates the counts for the observed words and word
 ; pairs.
 (define (observe-text plain-text)
-	(begin
-		(relex-parse plain-text)  ;; send plain-text to server
-		(update-link-counts (get-new-parsed-sentences))
-		(release-new-parsed-sents)
-		(delete-sentences)
-	)
+	(update-link-counts (nlp-parse plain-text))
```

Comment:
Oh, and one more note:  `(update-link-counts` above is counting word-pairs, which is something that your code does not do or need to do.  It also updates the counts in the SQL database, which, again, your code does not do, and probably(?) does not need to do.  In short, these are very different processing paths.